code/game/machinery/washing_machine.dm

```diff
@@ -37,65 +38,78 @@
-			
-				for(var/old_color in colors)
-					if(findtext(I.name, old_color))
-						I.name = n_replace(I.name, old_color, wash_color)
-						break
-
-				for(var/obj/item/clothing/J in I.contents)
-					J.color = wash_color
-					J.desc = J.desc + (length(J.desc) > 0 ? " " : "") + "It looks soaked in [wash_color] tincture."
+	if(locate(/obj/item/weapon/wrench) in contents)
+		flash_and_shake(src, "#FF0000", 2 , 100, xdif = 3, ydif = 3)
+		visible_message("<span class='danger'>[src] begins to shake violently!", "<span class='italics'>You hear various bangs</span>")
+	working = 1
+	spawn(200)
+
+		for(var/atom/A in contents)
+			A.clean_blood()
+
+			for(var/atom/B in A.contents)
+				B.clean_blood()
+
+		//Tanning!
+		for(var/obj/item/stack/sheet/hairlesshide/HH in contents)
+			var/obj/item/stack/sheet/wetleather/WL = new(src)
+			WL.amount = HH.amount
+			qdel(HH)
+
+		//Corgi costume says goodbye
+		for(var/obj/item/clothing/suit/hooded/ian_costume/IC in contents)
+			new /obj/item/weapon/reagent_containers/food/snacks/meat/slab/corgi(src)
+			qdel(IC)
+
+		if(crayon)
+			var/wash_color
+			if(istype(crayon,/obj/item/toy/crayon))
+				var/obj/item/toy/crayon/CR = crayon
+				wash_color = CR.colourName
+			else if(istype(crayon,/obj/item/weapon/stamp))
+				var/obj/item/weapon/stamp/ST = crayon
+				wash_color = ST.item_color
+
+			if(wash_color)
+				for(var/obj/item/clothing/I in contents)
+					I.color = wash_color //Simply recolor the items.
+					I.desc = I.desc + (length(I.desc) > 0 ? " " : "") + "It looks soaked in [wash_color] tincture."
+
+					var/list/colors = list("light brown", "brown", "cyan", "fingerless", "combat", "tactical", "yellowgreen", "darkred", "lightred", "maroon", "red", "orange", "rainbow", "lightgreen", "green", "lightpurple", "purple", "gold", "darkblue", "lightblue", "aqua", "blue", "yellow", "black", "grey", "gray", "white", "latex", "nitrile", "budget insulated", "insulated", "captain's")
 
 					for(var/old_color in colors)
-						if(findtext(J.name, old_color))
-							J.name = n_replace(J.name, old_color, wash_color)
+						if(findtext(I.name, old_color))
+							I.name = n_replace(I.name, old_color, wash_color)
 							break
-		qdel(crayon)
-		crayon = null
 
+					for(var/obj/item/clothing/J in I.contents)
+						J.color = wash_color
+						J.desc = J.desc + (length(J.desc) > 0 ? " " : "") + "It looks soaked in [wash_color] tincture."
 
-	if( locate(/mob,contents) )
-		state = 7
-		gibs_ready = 1
-	else
-		state = 4
-	update_icon()
+						for(var/old_color in colors)
+							if(findtext(J.name, old_color))
+								J.name = n_replace(J.name, old_color, wash_color)
+								break
+			qdel(crayon)
+			crayon = null
+
+		if(locate(/obj/item/weapon/wrench) in contents)
+			state = 1
+			for(var/atom/movable/O in contents)
+				O.loc = get_turf(src)
+				O.throw_at(get_edge_target_turf(src,pick(NORTH, SOUTH, EAST, WEST)), 500, 9, zone="chest")
+			update_icon()
+			working = 0
```

Comment:
No it's not, there's a return three lines down from this to prevent terrible bugs.